diff -urN src/imx_drv.c.orig src/imx_drv.c
--- src/imx_drv.c.orig	2010-03-22 18:07:25.000000000 +0000
+++ src/imx_drv.c	2010-07-25 09:30:11.000000000 +0000
@@ -148,67 +148,6 @@
 
 /* -------------------------------------------------------------------- */
 
-static const char *afbSymbols[] = {
-	"afbScreenInit",
-	"afbCreateDefColormap",
-	NULL
-};
-
-static const char *fbSymbols[] = {
-	"fbScreenInit",
-	"fbPictureInit",
-	NULL
-};
-
-static const char *shadowSymbols[] = {
-	"shadowAdd",
-	"shadowInit",
-	"shadowSetup",
-	"shadowUpdatePacked",
-	"shadowUpdatePackedWeak",
-	"shadowUpdateRotatePacked",
-	"shadowUpdateRotatePackedWeak",
-	NULL
-};
-
-static const char *fbdevHWSymbols[] = {
-	"fbdevHWInit",
-	"fbdevHWProbe",
-	"fbdevHWSetVideoModes",
-	"fbdevHWUseBuildinMode",
-
-	"fbdevHWGetDepth",
-	"fbdevHWGetLineLength",
-	"fbdevHWGetName",
-	"fbdevHWGetType",
-	"fbdevHWGetVidmem",
-	"fbdevHWLinearOffset",
-	"fbdevHWLoadPalette",
-	"fbdevHWMapVidmem",
-	"fbdevHWUnmapVidmem",
-
-	/* colormap */
-	"fbdevHWLoadPalette",
-	"fbdevHWLoadPaletteWeak",
-
-	/* ScrnInfo hooks */
-	"fbdevHWAdjustFrameWeak",
-	"fbdevHWEnterVTWeak",
-	"fbdevHWLeaveVTWeak",
-	"fbdevHWModeInit",
-	"fbdevHWRestore",
-	"fbdevHWSave",
-	"fbdevHWSaveScreen",
-	"fbdevHWSaveScreenWeak",
-	"fbdevHWSwitchModeWeak",
-	"fbdevHWValidModeWeak",
-
-	"fbdevHWDPMSSet",
-	"fbdevHWDPMSSetWeak",
-
-	NULL
-};
-
 #ifdef XFree86LOADER
 
 MODULESETUPPROTO(IMXSetup);
@@ -237,8 +176,6 @@
 	if (!setupDone) {
 		setupDone = TRUE;
 		xf86AddDriver(&IMX, module, HaveDriverFuncs);
-		LoaderRefSymLists(afbSymbols, fbSymbols,
-				  shadowSymbols, fbdevHWSymbols, NULL);
 		return (pointer)1;
 	} else {
 		if (errmaj) *errmaj = LDR_ONCEONLY;
@@ -311,8 +248,6 @@
 	if (!xf86LoadDrvSubModule(drv, "fbdevhw"))
 	    return FALSE;
 	    
-	xf86LoaderReqSymLists(fbdevHWSymbols, NULL);
-	
 	for (i = 0; i < numDevSections; i++) {
 
 	    dev = xf86FindOptionValue(devSections[i]->options,"imx");
@@ -359,7 +294,6 @@
 	IMXPtr fPtr;
 	int default_depth, fbbpp;
 	const char *mod = NULL, *s;
-	const char **syms = NULL;
 	int type;
 
 	if (flags & PROBE_DETECT) return FALSE;
@@ -539,7 +473,6 @@
 		case 24:
 		case 32:
 			mod = "fb";
-			syms = fbSymbols;
 			break;
 		default:
 			xf86DrvMsg(pScrn->scrnIndex, X_ERROR,
@@ -572,10 +505,6 @@
 		IMXFreeRec(pScrn);
 		return FALSE;
 	}
-	if (mod && syms) {
-		xf86LoaderReqSymLists(syms, NULL);
-	}
-
 	/* Perform EXA pre-init */
 	if (fPtr->useAccel) {
 
@@ -597,7 +526,6 @@
 			IMXFreeRec(pScrn);
 			return FALSE;
 		}
-		xf86LoaderReqSymLists(shadowSymbols, NULL);
 	}
 
 	TRACE_EXIT("PreInit");
diff -urN xf86-video-imx-10.05.02-orig/src/imx_exa_z160.c xf86-video-imx-10.05.02/src/imx_exa_z160.c
--- src/imx_exa_z160.c	2010-03-22 18:07:25.000000000 +0000
+++ src/imx_exa_z160.c	2010-07-25 09:30:11.000000000 +0000
@@ -125,27 +125,6 @@
 
 #define IMXEXAPTR(p) ((IMXEXAPtr)((p)->exaDriverPrivate))
 
-#if !IMX_EXA_ENABLE_EXA_INTERNAL
-
-/* Function symbols dynamically loaded from EXA module. */
-static const char *exaSymbols[] = {
-	"exaDriverAlloc",
-	"exaDriverInit",
-	"exaDriverFini",
-	"exaOffscreenAlloc",
-	"exaOffscreenFree",
-	"exaGetPixmapOffset",
-	"exaGetPixmapDriverPrivate",
-	"exaGetPixmapPitch",
-	"exaGetPixmapSize",
-	"exaGetDrawablePixmap",
-	"exaMarkSync",
-	"exaWaitSync",
-	NULL
-};
-
-#endif
-
 /* Prototype for function not defined in exa.h */
 extern PixmapPtr exaGetDrawablePixmap(DrawablePtr pDrawable);
 
@@ -2041,7 +2020,6 @@
 	}
 
 	/* Load required EXA symbols */
-	xf86LoaderReqSymLists(exaSymbols, NULL);
 #endif
 
 	/* initialize state of Z160 data structures */
